titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/JobManagerUtil.java

```diff
@@ -133,21 +133,31 @@ public static String getCapacityGroupDescriptorName(JobDescriptor<?> jobDescript
             }
 
             Task newTask = JobFunctions.changeTaskStatus(oldTask, newTaskStatus);
-            Task newTaskWithPlacementData = attachTitusExecutorData(newTask, detailsOpt);
+            Task newTaskWithPlacementData = attachTitusExecutorNetworkData(newTask, detailsOpt);
             return Optional.of(newTaskWithPlacementData);
         };
     }
 
-    public static Task attachTitusExecutorData(Task task, Optional<TitusExecutorDetails> detailsOpt) {
+    public static Task attachTitusExecutorNetworkData(Task task, Optional<TitusExecutorDetails> detailsOpt) {
         return detailsOpt.map(details -> {
             TitusExecutorDetails.NetworkConfiguration networkConfiguration = details.getNetworkConfiguration();
             if (networkConfiguration != null) {
+                String ipv4 = networkConfiguration.getIpAddress();
+                String ipv6 = networkConfiguration.getIpV6Address();
+                String primaryIP = networkConfiguration.getPrimaryIpAddress();
                 Map<String, String> newContext = new HashMap<>(task.getTaskContext());
                 BiConsumer<String, String> contextSetter = (key, value) -> StringExt.applyIfNonEmpty(value, v -> newContext.put(key, v));
 
-                contextSetter.accept(TaskAttributes.TASK_ATTRIBUTES_CONTAINER_IP, networkConfiguration.getIpAddress());
-                contextSetter.accept(TaskAttributes.TASK_ATTRIBUTES_CONTAINER_IPV4, networkConfiguration.getIpAddress());
-                Evaluators.acceptNotNull(networkConfiguration.getIpV6Address(), ipv6 -> newContext.put(TaskAttributes.TASK_ATTRIBUTES_CONTAINER_IPV6, ipv6));
+                if (networkConfiguration.getNetworkMode() == "Ipv6AndIpv4Fallback") {
```

Comment:
`equals`?
Use constant for `Ipv6AndIpv4Fallback`?

Reply:
Got it.
I think I've imported the constant correctly, please double check that importing the enum like I have is appropriate.